mineral-ui/mineral-ui
# [Checkbox] IE/Edge toggle indeterminate state on single click

### Description

* Add support for controlled/uncontrolled indeterminate checkbox state
* Fix IE/Edge toggle indeterminate state on single click
* This update fixes the Select all checkbox in selectable Tables
* Make Table density and messages props optional to avoid propType warning

### Motivation and context

closes #739 

### Screenshots, videos, or demo, if appropriate

https://739-checkbox--mineral-ui.netlify.com/

### How to test

1. View the [selectable table](https://739-checkbox--mineral-ui.netlify.com/components/table/selectable) example in IE11 or Edge.
2. Toggle the Select All checkbox and note that it toggles correctly on a single click
3. Also note the [indeterminate checkbox](https://739-checkbox--mineral-ui.netlify.com/components/checkbox/indeterminate) example

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

    * Fixes issue where indeterminate checkboxes in IE/Edge would take 2 clicks to toggle initially

- New feature (non-breaking change which adds functionality)

    * Add support for controlled and uncontrolled indeterminate checkboxes

### Checklist
